src/audio/codec_adapter/codec_adapter.c

```diff
@@ -556,7 +556,10 @@ void codec_adapter_free(struct comp_dev *dev)
 	if (ret)
 		comp_err(dev, "codec_adapter_free(): error %d, codec free failed", ret);
 
-	buffer_free(mod->local_buff);
+	/* if module is not prepared, local_buffer won't be allocated */
+	if (mod->local_buff)
```

Comment:
I would argue the proper fix is a
```
if (!buffer)
  return
```
in the free call like most other implementations

Reply:
Make sense.